src/ansi-c/c_typecheck_code.cpp

```diff
@@ -102,9 +103,24 @@ void c_typecheck_baset::typecheck_code(codet &code)
   }
   else if(statement==ID_static_assert)
   {
-    assert(code.operands().size()==2);
+    PRECONDITION(code.operands().size() == 2);
+
     typecheck_expr(code.op0());
     typecheck_expr(code.op1());
+
+    code.op0() = typecast_exprt::conditional_cast(code.op0(), bool_typet{});
```

Comment:
That might not work, consider `implicit_typecast_bool`?

Reply:
Done!